File: berlin-clock-play/app/controllers/ApiClockController.scala
package controllers

import java.util.Calendar
import javax.inject._

import akka.actor.ActorSystem
import model.BerlinTimeModel
import play.api.libs.json.Json
import play.api.mvc._
import services.TimeRepresentationService

import scala.concurrent.{ExecutionContext, Future}


@Singleton
class ApiClockController @Inject()(cc: ControllerComponents, timeRepresentationService: TimeRepresentationService, actorSystem: ActorSystem)(implicit exec: ExecutionContext) extends AbstractController(cc) {

  def getBerlinTimeRepresentation = Action.async {
    getFutureTimeRepresentation(Calendar.getInstance())
      .map { result => Ok(Json.toJson(result)) }
  }


  private def getFutureTimeRepresentation(timeToRepresent: Calendar) = Future[BerlinTimeModel] {
    timeRepresentationService.getBerlinRepresentation(timeToRepresent)
  }
}

File: berlin-clock-play-java/app/controllers/ApiClockController.java
package controllers;

import models.BerlinTimeModel;
import play.libs.Json;
import play.mvc.Controller;

import javax.inject.Inject;
import javax.inject.Singleton;

import play.mvc.Result;
import services.TimeRepresentationService;
import java.util.Calendar;

@Singleton
public class ApiClockController extends Controller {

    @Inject
    private TimeRepresentationService service;


    public Result getBerlinTimeRepresentation() {
        BerlinTimeModel representation = service.getBerlinRepresentation(Calendar.getInstance());

        return ok(Json.toJson(representation));
    }
}
